pkgs/applications/version-management/gittyup/default.nix

```
@@ -0,0 +1,111 @@
+{ lib
+, mkDerivation
+, fetchFromGitHub
+, cmark
+, mount
+, http-parser
+, perl
+, cmake
+, qtbase
+, qttools
+, git
+, git-lfs
+, ninja
+, libssh2
+, libgit2
+, pkg-config
+, libsecret
+, libgnome-keyring
+, pcre
+, wrapQtAppsHook
+}:
+mkDerivation rec {
+  pname = "gittyup";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "Murmele";
+    repo = "Gittyup";
+    rev = "gittyup_v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-WJzw1uFCE5dyI1F0FEzww05zCsYPEgeXBz+HogTYX9w=";
+  };
+
+  buildInputs = [
+    qtbase
+    git
+    git-lfs
+    libssh2
+    libgit2
+    libsecret
+    libgnome-keyring
+    pcre
+  ];
+
+  nativeBuildInputs = [
+    cmark
+    mount
+    http-parser
+    perl
+    cmake
+    qttools
+    ninja
+    pkg-config
+  ];
+
+  cmakeFlage = [
+    "-W no-dev"
+    "-DCMAKE_BUILD_TYPE=Release"
+    "-DCMAKE_PREFIX_PATH=${qtbase}"
+    "-DENABLE_REPRODUCIBLE_BUILDS=ON"
+    "-DBUILD_SHARED_LIBS=OFF"
+  ];
+
+  postInstall = ''
+    shopt -s extglob
+    mkdir -p $out/temp
+    mv $out/!(temp) $out/temp
+    mkdir -p $out/lib
+    mv $out/temp $out/lib/gittyup
+
+    rm -rf "$out/lib/gittyup/"*.so.*
+    rm -rf "$out/lib/gittyup/Plugins"
+    rm -rf "$out/lib/gittyup/lib"
+    rm -rf "$out/lib/gittyup/lib64"
+    rm -rf "$out/lib/gittyup/include"
+    rm -rf "$out/lib/gittyup/share"
+
+    install -d "$out/bin"
+    ln -s $out/lib/gittyup/Gittyup "$out/bin/gittyup"
+    chmod 0555 "$out/bin/gittyup"
+
+    install -Dm644 $src/LICENSE.md "$out/share/licenses/${pname}/LICENSE"
```

```suggestion
    install -Dm644 $src/LICENSE.md "$out/share/licenses/gittyup/LICENSE"
```
Also we don't really need a coppy of the mit license.